test/unittest/TrafficInjector/TrafficInjectorTests.cpp

```diff
@@ -0,0 +1,311 @@
+        }
+
+        {
+            std::string topic_name = "_fastdds_statistics_edp_packets";
+            TypeSupport type(new eprosima::fastdds::statistics::EntityCountPubSubType());
+            type.register_type(participant_);
+
+            Topic* topic = participant_->create_topic(topic_name,
+                            type->getName(), eprosima::fastdds::dds::TOPIC_QOS_DEFAULT);
+            topics_.insert(topic);
+            readers_.insert(subscriber_->create_datareader(topic, eprosima::fastdds::dds::DATAREADER_QOS_DEFAULT,
+                    &listener_));
+        }
+
+        {
+            std::string topic_name = "_fastdds_statistics_discovered_entity";
+            TypeSupport type(new eprosima::fastdds::statistics::DiscoveryTimePubSubType());
+            type.register_type(participant_);
+
+            Topic* topic = participant_->create_topic(topic_name,
+                            type->getName(), eprosima::fastdds::dds::TOPIC_QOS_DEFAULT);
+            topics_.insert(topic);
+            readers_.insert(subscriber_->create_datareader(topic, eprosima::fastdds::dds::DATAREADER_QOS_DEFAULT,
+                    &listener_));
+        }
+
+        {
+            std::string topic_name = "_fastdds_statistics_sample_datas";
+            TypeSupport type(new eprosima::fastdds::statistics::SampleIdentityCountPubSubType());
+            type.register_type(participant_);
+
+            Topic* topic = participant_->create_topic(topic_name,
+                            type->getName(), eprosima::fastdds::dds::TOPIC_QOS_DEFAULT);
+            topics_.insert(topic);
+            readers_.insert(subscriber_->create_datareader(topic, eprosima::fastdds::dds::DATAREADER_QOS_DEFAULT,
+                    &listener_));
+        }
+
+        {
+            std::string topic_name = "_fastdds_statistics_physical_data";
+            TypeSupport type(new eprosima::fastdds::statistics::PhysicalDataPubSubType());
+            type.register_type(participant_);
+
+            Topic* topic = participant_->create_topic(topic_name,
+                            type->getName(), eprosima::fastdds::dds::TOPIC_QOS_DEFAULT);
+            topics_.insert(topic);
+            readers_.insert(subscriber_->create_datareader(topic, eprosima::fastdds::dds::DATAREADER_QOS_DEFAULT,
+                    &listener_));
+        }
+    }
+
+    void TearDown()
+    {
+        for (auto reader : readers_)
+        {
+            subscriber_->delete_datareader(reader);
+        }
+
+        for (auto topic : topics_)
+        {
+            participant_->delete_topic(topic);
+        }
+
+        participant_->delete_subscriber(subscriber_);
+
+        DomainParticipantFactory::get_instance()->delete_participant(participant_);
+    }
+
+};
+
+TEST_F(TrafficInjectorTests, Basic)
+{
+    std::string traffic_dump = "traffic_dump.json";
+    TrafficInjector tf;
+    tf.load(traffic_dump);
+
+
+    while (tf.inject_next_message())
+    {
+
+    }
```

Comment:
Should we check that the messages are received?

Reply:
ef359fa addresses this. It checks that at least one message for every topic has been received. As the dump contains a single message on every topic, this is enough to check that all messages are received and processed by the correct writer. However, I skipped checking that the fields of the received messages are correct, as this would require parsing the json file... which is what the injector does anyways.